storage/innobase/srv/srv0start.cc

```diff
@@ -1123,11 +1123,15 @@ static dberr_t srv_undo_tablespaces_open(bool backup_mode) {
   for (space_id_t num = 1; num <= FSP_MAX_UNDO_TABLESPACES; ++num) {
     /* Check if this undo tablespace was in the process of being truncated.
     If so, recreate it and add it to the construction list. */
-    dberr_t err = srv_undo_tablespace_fixup_num(num);
-    if (err != DB_SUCCESS) {
-      undo::spaces->x_unlock();
-      return (err);
+#ifdef XTRABACKUP
+    if (!backup_mode) {
+      dberr_t err = srv_undo_tablespace_fixup_num(num);
```

Comment:
this is required for lock-ddl true as well. Because of existing bug in MySQL, there are left over _trunc.log files.( I could not reproduce this case  but as pointed by user It is possible) Also it wrong  if PXB tries to create any file on server directory 

Backup fails for concurrent undo truncation happens. [PXB-2255](https://jira.percona.com/browse/PXB-2255)